rasa/shared/core/domain.py

```diff
@@ -317,7 +317,11 @@ def merge_domain_dicts(
         if override or domain2.get("session_config"):
             combined[SESSION_CONFIG_KEY] = domain_dict[SESSION_CONFIG_KEY]
 
-        if combined[KEY_INTENTS] or domain_dict[KEY_INTENTS]:
+        if combined.get(KEY_INTENTS) or domain_dict.get(KEY_INTENTS):
+            if not combined.get(KEY_INTENTS):
+                combined[KEY_INTENTS] = []
+            if not domain_dict.get(KEY_INTENTS):
+                domain_dict[KEY_INTENTS] = []
             combined[KEY_INTENTS] = self.merge_lists_of_dicts(
                 combined[KEY_INTENTS], domain_dict[KEY_INTENTS], override
```

Comment:
Is there a reason why we shouldn't be setting a default here (ex: 'combined.get('KEY_INTENTS', [])') instead of setting them inside conditionals?

Reply:
So, `combined` or `domain_dict` may or may not have any `'intents'`. This depends on the domain yml file that they are derived from. Is there a neater way to assign a default value if no key exists (other than what @jupyterjazz has suggested above?)

Reply:
Ah, that's fine then. That would be trick then, stylistic brevity aside.